### PR TITLE
fix(webapp): reset delete document dialog

### PR DESCRIPTION
**Pull request by ubinatus**
This PR makes a small but useful tweak to the `DeleteDocumentDialog`. Now, the input field gets cleared whenever the dialog is opened. Here’s what’s changed:

1. **Clear Field After Deleting**: After you delete something and open the dialog again, it won’t show the old, deleted text anymore. It’s clean and ready for the next delete.

2. **Type Again to Confirm**: If you type something but close the dialog without deleting, you’ll have to type it again next time. This way, it makes sure the user really mean to delete something and didn't do it by mistake.

Demo Link:
See the old vs. new in action here: https://www.loom.com/share/80eca0d3b1994f7cbcab6f222db2dbfe?sid=ebc6135c-345e-4640-b395-daff190a96e7

It’s a small change, but it makes the delete process safer and smoother.

### CODE REVIEW

**Comment by Mythie** [comment]
Hey there, it looks like you haven't accepted our contributor license agreement yet. In order for us to accept your pull request we ask that you please fill out the CLA:

https://documen.so/cla

**Comment by catalinpit** [comment]
Hey @ubinatus, thanks for the PR! The demo link doesn't work. Could you update the link, please?

**Comment by ubinatus** [comment]
Hi @catalinpit! I've updated the description with a working public loom url. Sorry about that, it was toggled as a private link.

**Comment by catalinpit** [comment]
> Hi @catalinpit! I've updated the description with a working public loom url. Sorry about that, it was toggled as a private link.

Hey, thank you! In the meanwhile, I tested the changes and I came across an issue. If you try to delete a draft document, the `delete` button is disabled. See the attached pic.

CleanShot 2023-12-19 at 10 57 19

That's because in the `useEffect` we set `isDeleteEnabled` to false, even if the document is a draft.  Perhaps, we should do a check to see if the status of the document is set to draft before setting `isDeleteEnabled` to false? Just an idea. 👀

File: apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx [hunk 1]
```diff
@@ -50,6 +50,13 @@ export const DeleteDocumentDialog = ({
     },
   });
 
+  useEffect(() => {
+    if (open) {
+      setInputValue('');
+      setIsDeleteEnabled(false);
```
Comment (catalinpit) [inline apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx:56]: This line won't allow users to delete draft documents. See the main comment here https://github.com/documenso/documenso/pull/762#issuecomment-1862367650 🙏

**Review by catalinpit — CHANGES_REQUESTED** [review]

File: apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx (hunk 1 above)
Comment (ubinatus) [reply apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx]: Updated! Now when deleting draft documents the button not disabled.

**Review by catalinpit — DISMISSED** [review]
Looks good. Thanks for the work!

File: apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx [hunk 2]
```diff
@@ -50,6 +50,13 @@ export const DeleteDocumentDialog = ({
     },
   });
 
+  useEffect(() => {
+    if (open) {
+      setInputValue('');
+      setIsDeleteEnabled(status === DocumentStatus.DRAFT ? true : false);
```
Comment (dguyen) [inline apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx:56]: Can just be:

```
setIsDeleteEnabled(status === DocumentStatus.DRAFT);
```

Comment (ubinatus) [reply apps/web/src/app/(dashboard)/documents/delete-document-dialog.tsx]: Agree! No need for that ternary. Done!

**Review by dguyen — APPROVED** [review]
Looks good to me, thanks 😃

**Pull request merged**